internal/engine/client.go

```diff
@@ -16,18 +16,32 @@ import (
 	_ "github.com/moby/buildkit/client/connhelper/podmancontainer" // import the podman connection driver
 )
 
-func Client(ctx context.Context, remote *url.URL) (*bkclient.Client, error) {
+// Client returns a buildkit client, whether privileged execs are enabled, or an error
+func Client(ctx context.Context, remote *url.URL) (*bkclient.Client, bool, error) {
 	buildkitdHost := remote.String()
 	if remote.Scheme == DockerImageProvider {
 		var err error
 		buildkitdHost, err = dockerImageProvider(ctx, remote)
 		if err != nil {
-			return nil, err
+			return nil, false, err
 		}
 	}
 
-	if err := waitBuildkit(ctx, buildkitdHost); err != nil {
-		return nil, err
+	workerInfo, err := waitBuildkit(ctx, buildkitdHost)
+	if err != nil {
+		return nil, false, err
+	}
+	var privilegedExecEnabled bool
+	if len(workerInfo) > 0 {
+		for k, v := range workerInfo[0].Labels {
+			// TODO:(sipsma) we set this custom label in the default engine config
+			// toml. It's not the best solution but the only way to get this
+			// information to the client right now.
+			if k == "privilegedEnabled" && v == "true" {
```

Comment:
:+1: - neat idea; I had a similar problem in #4711 where the schema needed to know the DNS domain. I decided to just remove it but this makes sense too, and might be useful information to make available to an operator if we decide to keep it after all.